src/app/views/validation/results/results.component.js

```diff
@@ -8,11 +8,16 @@ class ResultsController {
   }
 
   downloadFastqFiles() {
+    this.loading = true;
     if (!this.expeditionCode) {
       return;
     }
 
-    this.DataService.generateSraData(this.projectId, this.expeditionCode);
+    this.DataService.generateSraData(this.projectId, this.expeditionCode).then(
```

Comment:
we should replace `.then` with `.finally` so if an error occurs, we don't continue to show the loading symbol